File: src/main/java/de/ingrid/igeserver/exports/iso/CharacterString.java
package de.ingrid.igeserver.exports.iso;

import javax.xml.bind.annotation.XmlElement;

public class CharacterString {
	
	@XmlElement(name="CharacterString", namespace="http://www.isotc211.org/2005/gco")
	public String text;

	public CharacterString() {}
	
	public CharacterString(String text) {
		this.text = text;
	}

}

File: server/src/main/java/de/ingrid/igeserver/exports/iso/CharacterString.kt
package de.ingrid.igeserver.exports.iso

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty

data class CharacterString(
    @JacksonXmlProperty(localName="CharacterString", namespace="http://www.isotc211.org/2005/gco")
    private var text: String? = null,
    @JacksonXmlProperty(localName="Anchor", namespace="http://www.isotc211.org/2005/gmx")
    private var anchor: Anchor? = null
) {
    val value = text ?: anchor?.value
    val anchorHref = anchor?.href
    val isAnchor = anchor != null
}
